### PR TITLE
docs: add note about using regexp literals in queries

### PR DESCRIPTION
**Pull request by kamiyo**
Document regexp query filter to inform users about passing the regexp object directly as the field value instead of using $re.

### CODE REVIEW

File: docs/docs/query-conditions.md [hunk 1]
```diff
@@ -74,6 +74,16 @@ const res = await orm.em.find(Author, [1, 2, 7]);
...
+## Regular Expressions
+
+The `$re` operator takes a string as input value, and by default uses the case-sensitive operator. If you would like to use a `RegExp` object, i.e. to be able to set flags, then search directly on the field name without using the operator:
+
+```ts
+const res = await orm.em.find(Painter, {
+  lastName: /m[oa]net/i        // or new RegExp('m[oa]net', 'i')
+});
+```
+
```
Comment (rubiin) [inline docs/docs/query-conditions.md:86]: remove the extra line

Comment (B4nan) [reply docs/docs/query-conditions.md:86]: this one is correct, there should be blank lines around the headings

File: docs/docs/query-conditions.md [hunk 2]
```diff
@@ -74,6 +74,16 @@ const res = await orm.em.find(Author, [1, 2, 7]);
 | `$not`   | Inverts the effect of a query expression and returns documents that do not match the query expression.  |
 | `$or`    | Joins query clauses with a logical OR returns all documents that match the conditions of either clause. |
 
+## Regular Expressions
+
+The `$re` operator takes a string as input value, and by default uses the case-sensitive operator. If you would like to use a `RegExp` object, i.e. to be able to set flags, then search directly on the field name without using the operator:
+
+```ts
+const res = await orm.em.find(Painter, {
+  lastName: /m[oa]net/i        // or new RegExp('m[oa]net', 'i')
```
Comment (B4nan) [inline docs/docs/query-conditions.md:83]:
```suggestion
const res = await em.find(Painter, {
  lastName: /m[oa]net/i, // or `new RegExp('m[oa]net', 'i')`
```

**Review by B4nan — APPROVED** [review]
thanks!

**Pull request merged**